### PR TITLE
Add test case for keyed array item

### PR DESCRIPTION
**Pull request by jon-ht**
Not sure about test case naming, tell me if you have a better one :)

Part of #608

```
In YamlSourceManipulator.php line 549:
                                
  Cannot find the key "Bearer"  
                   
```

### CODE REVIEW

File: src/Util/YamlSourceManipulator.php [hunk 1]
```diff
@@ -623,7 +643,7 @@ private function getEndOfPreviousKeyPosition($key): int
 
     private function getKeyRegex($key)
     {
-        return sprintf('#%s( )*:#', preg_quote($key));
+        return sprintf('#(- )?%s( )*:#', preg_quote($key));
```
Comment (weaverryan) [inline src/Util/YamlSourceManipulator.php:646]: We either have too many changes in this PR, or we're missing some more tests :).

If I revert this change AND revert the changes from lines 530-538, the new test PASSES. It seems that only the code from 559-568 is actually needed. If the other code fixes some other bug, we need a test case for that.

Comment (jon-ht) [reply src/Util/YamlSourceManipulator.php]: You're right. I think this line was a "first resolution try" and I forgot it :)

**Comment by weaverryan** [comment]
Thanks @jon-ht!
